common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleClient.java

```diff
@@ -43,6 +44,18 @@
    *                        they will be kept in memory.
    */
   public abstract void fetchBlocks(
+      String host,
+      int port,
+      String execId,
+      int shuffleGenerationId,
+      String[] blockIds,
+      BlockFetchingListener listener,
+      DownloadFileManager downloadFileManager);
+
+  /**
+   * Fetch a sequence of non-shuffle blocks from a remote node asynchronously.
+   */
+  public abstract void fetchDataBlocks(
```

Comment:
@cloud-fan During I do the refactor of splitting ShuffleFetcher and NonShuffleFetcher and fix the UT, I realize that after SPARK-27677, the ExternalShuffleClient not only handle shuffle block but also handle local disk persist block, in the latest commit I just split them into 2 functions, but it's maybe strange to  handle non-shuffle block in a `ShuffleClient`, WDYT?

Reply:
This is a little weird, but according to  SPARK-27677, this is by design: we want external shuffle service to serve cached RDD blocks. Does normal shuffle also use `ShuffleClient`?

Reply:
maybe we should rename external shuffle service to external block store service.

Reply:
```
Does normal shuffle also use ShuffleClient?
```
Normal shuffle uses BlockTransferService, which is a subclass of ShuffleClient.
```
maybe we should rename external shuffle service to external block store service.
```
Thanks for the confirmation, let me think about it.